mesonbuild/build.py

```diff
@@ -915,9 +915,12 @@ def get_arglist(self):
     def process_files(self, name, files, state, extra_args=[]):
         output = GeneratedList(self, extra_args=extra_args)
         for f in files:
-            if not isinstance(f, str):
-                raise InvalidArguments('{} arguments must be strings.'.format(name))
-            output.add_file(os.path.join(state.subdir, f))
+            if isinstance(f, str):
+                output.add_file(os.path.join(state.subdir, f))
```

Comment:
It would be nice to convert the string to a File object here. This way the list contains only File objects and we don't need the type checks (str or File) in the backends. This would simplify things nicely.